lib/drip/client.rb

```diff
@@ -55,56 +58,65 @@ def content_type
     end
 
     def get(url, options = {})
-      make_request(:get, url, options)
+      make_request(Net::HTTP::Get, make_uri(url), options)
     end
 
     def post(url, options = {})
-      make_request(:post, url, options)
+      make_request(Net::HTTP::Post, make_uri(url), options)
     end
 
     def put(url, options = {})
-      make_request(:put, url, options)
+      make_request(Net::HTTP::Put, make_uri(url), options)
     end
 
     def delete(url, options = {})
-      make_request(:delete, url, options)
+      make_request(Net::HTTP::Delete, make_uri(url), options)
     end
 
-    def make_request(verb, url, options)
+  private
+
+    def make_uri(path)
+      URI(url_prefix) + URI(path)
+    end
+
+    def make_request(verb_klass, uri, options, step = 0)
+      raise TooManyRedirectsError, 'too many HTTP redirects' if step >= REDIRECT_LIMIT
+
       build_response do
-        connection.send(verb) do |req|
-          req.url url
+        Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == "https") do |http|
+          if verb_klass.is_a? Net::HTTP::Get
+            uri.query = URI.encode_www_form(options)
+          end
+
+          request = verb_klass.new uri
+
+          unless verb_klass.is_a? Net::HTTP::Get
+            request.body = options.to_json
+          end
 
-          if verb == :get
-            req.params = options
+          request['User-Agent'] = "Drip Ruby v#{Drip::VERSION}"
+          request['Content-Type'] = content_type
+          request['Accept'] = "*/*"
+
+          if access_token
+            request['Authorization'] = "Bearer #{access_token}"
```

Comment:
Is there a standard for quotes vs double-quotes in this project?

Reply:
Not at this point, it looks like. It's probably worth doing rubocop cleanups.